src/components/util/Rivet.jsx

```diff
@@ -1,11 +1,139 @@
+const TypescaleNum = [12, 14, 18, 20, 23, 26, 29, 32, 36, 41, 46, 52];
+const TypescaleNumStr = [
+  "12",
+  "14",
+  "18",
+  "20",
+  "23",
+  "26",
+  "29",
+  "32",
+  "36",
+  "41",
+  "46",
+  "52",
+];
+const TypescaleSize = ["xxs", "xs", "base", "sm", "md", "lg", "xl", "xxl"];
+const TypescaleSizes = [...TypescaleNum, ...TypescaleNumStr, ...TypescaleSize];
+const TypescaleResp = ["sm-up", "md-up", "lg-up", "xl-up", "xxl-up"]
+  .map((r) => TypescaleSizes.map((s) => `${s}-${r}`))
+  .flat(1);
+const Typescale = [...TypescaleSizes, ...TypescaleResp];
+const FontFamily = ["sans", "serif", "mono"];
+const TextAlign = ["left", "center", "right"];
+const WidthSize = [
+  "xxs",
+  "xs",
+  "base",
+  "sm",
+  "md",
+  "lg",
+  "xl",
+  "xxl",
+  "3-xl",
+  "4-xl",
+];
+const WidthResp = ["sm-up", "md-up", "lg-up", "xl-up", "xxl-up"]
+  .map((r) => WidthSize.map((s) => `${s}-${r}`))
+  .flat(1);
+const Width = [...WidthSize, ...WidthResp];
+const ZIndexNum = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+const ZIndexStr = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10"];
+const ZIndex = [...ZIndexNum, ...ZIndexStr];
+const Display = ["inline", "inline-block", "block", "none"];
+const FlexResp = ["sm-up", "md-up", "lg-up", "xl-up", "xxl-up"];
+const FlexInline = ["inline", ...FlexResp.map((r) => `inline-${r}`)];
+const FlexNormal = ["normal", ...FlexResp.map((r) => `normal-${r}`)];
+const FlexDirectionBase = ["row", "row-reverse", "column", "column-reverse"];
+const FlexDirectionResp = FlexResp.map((r) =>
+  FlexDirectionBase.map((s) => `${s}-${r}`)
+).flat(1);
+const FlexDirection = [...FlexDirectionBase, ...FlexDirectionResp];
+const FlexWrapBase = ["wrap", "no-wrap", "wrap-reverse"];
+const FlexWrapResp = FlexResp.map((r) =>
+  FlexWrapBase.map((s) => `${s}-${r}`)
+).flat(1);
+const FlexWrap = [...FlexWrapBase, ...FlexWrapResp];
+const JustifyBase = [
+  "start",
+  "end",
+  "center",
+  "space-between",
+  "space-around",
+  "space-evenly",
+];
+const JustifyResp = FlexResp.map((r) =>
+  JustifyBase.map((s) => `${s}-${r}`)
+).flat(1);
+const JustifyContent = [...JustifyBase, ...JustifyResp];
+const AlignBase = ["start", "end", "center", "stretch", "baseline"];
+const AlignResp = FlexResp.map((r) => AlignBase.map((s) => `${s}-${r}`)).flat(
+  1
+);
+const AlignContent = [...AlignBase, ...AlignResp];
+const AlignItems = [...AlignBase, ...AlignResp];
+const AlignSelfBase = [
+  "self-start",
+  "self-end",
+  "center-end",
+  "stretch-end",
+  "baseline-end",
+];
+const AlignSelfResp = FlexResp.map((r) =>
+  AlignSelfBase.map((s) => `${s}-${r}`)
+).flat(1);
+const AlignSelf = [...AlignSelfBase, ...AlignSelfResp];
+const FlexShrinkGrowBase = ["0", "1"];
+const FlexShrinkGrowResp = FlexResp.map((r) =>
+  FlexShrinkGrowBase.map((s) => `${s}-${r}`)
+).flat(1);
+const FlexShrinkGrow = [0, 1, ...FlexShrinkGrowBase, ...FlexShrinkGrowResp];
+const Weight = ["regular", "medium", "bold"];
```

Comment:
These constants (and the support functions later) could probably be extracted to a separate file and imported here to make it easier to read.